src/fides/api/models/privacy_request.py

```diff
@@ -451,6 +460,15 @@ def delete(self, db: Session) -> None:
             provided_identity.delete(db=db)
         super().delete(db=db)
 
+    def soft_delete(self, db: Session, user_id: str) -> None:
+        """
+        Soft delete the privacy request, marking it as deleted and setting the user who deleted it.
+        """
+        self.deleted_at = datetime.utcnow()
+        self.deleted_by = user_id
+        self.save(db)
+        # TODO: do we need to remove cache keys?
+
```

Comment:
the question is because the "real" `delete` method (defined right above) does the following: 
```py
        cache: FidesopsRedis = get_cache()
        all_keys = get_all_cache_keys_for_privacy_request(privacy_request_id=self.id)
        for key in all_keys:
            cache.delete(key)
        for provided_identity in self.provided_identities:  # type: ignore[attr-defined]
            provided_identity.delete(db=db)
```